01/frontend/src/components/App.tsx

```diff
@@ -1,16 +1,28 @@
-import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
-import Form from "./Form";
-import LoggedIn from "./LoggedIn";
+import { useEffect } from "react";
+import { Route, Routes } from "react-router-dom";
+import Form from "@components/Form";
+import LoggedIn from "@components/LoggedIn";
+import useAuth from "@hooks/useAuth";
 
 const App = () => {
+  const { authenticate, isAuthenticated } = useAuth();
+
+  useEffect(() => {
+    if (localStorage.getItem("jwtToken") !== null) {
+      authenticate();
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
   return (
-    <Router>
-      <Routes>
+    <Routes>
+      {!isAuthenticated ? (
```

Comment:
Ok. You can create two arrays with protected routes and public routes for better readability.

it could look something like that:
```ts
const protectedRoutes = [
    {path: "/api/admin/dashboard", component: <AdminPanel/>},
    {path: "/api/user/dashboard", component: <UserInfo/>}
];

const publicRoutes = [
  { path: "api/user/login", component: <Login /> },
  { path: "api/user/logout", component: <Logout /> },
];

```

and in App.tsx you could make something like that:

```ts
<Routes>
          {publicRoutes.map((route, idx) => (
            <Route path={route.path} key={idx} element={route.component} />
          ))}
          
           {protectedRoutes.map((route, idx) => (
            <Route
              path={route.path}
              key={idx}
              element={
                <React.Fragment>
                  <AuthProvider>
                    {route.component}
                  </AuthProvider>
                </React.Fragment>
              }
            />
          ))}
</Routes>

```
Create AuthProvider component and inside this component check, if user is authenticated

Reply:
Also in AuthProvider you can implement auto logout